src/main/webapp/3Dscript.js

```diff
@@ -44,140 +94,340 @@ function initThreeJs() {
+  xCylinder.instanceMatrix.setUsage(THREE.DynamicDrawUsage);
+  scene.add(xCylinder);
+
+  return [xCylinder, yCylinder, zCylinder];
+}
+/**
+  * Converts latitude, longitude, and altitude coordinates (LLA) to
+  * Earth-Centered, Earth-Fixed coordinates (ECEF).
+  * @param {number} lat Latitude represented in Decimal Degrees
+  * @param {number} lng Longitude represented in Decimal Degrees
+  * @param {number} alt Altitude represented in Decimal Degrees
+  * @return {array} An array containing x, y, z ECEF coordinates
+  *     all of which are in meters.
+  */
+function llaToEcef(lat, lng, alt) {
+  // LLA to ECEF Paper: https://microem.ru/files/2012/08/GPS.G1-X-00006.pdf
+  const majorAxis = 6378137; // Meters
+  const flatConstant = 1 / 298.25722356; // Scalar
+  const minorAxis = majorAxis*(1-flatConstant); // Meters
+
+  // Eccentricity.
+  const eSq = (Math.pow(majorAxis, 2) -
+    Math.pow(minorAxis, 2)) / Math.pow(majorAxis, 2);
+  const lambda = THREE.Math.degToRad(lat);
+  const phi = THREE.Math.degToRad(lng);
+
+  // Radius of curvature on the Earth
+  const radius = majorAxis /
+    Math.sqrt(1 - eSq * Math.pow(Math.sin(lambda), 2));
+  // Conversion equations.
+  const x = (radius + alt) * Math.cos(lambda) * Math.cos(phi);
+  const y = (radius + alt) * Math.cos(lambda) * Math.sin(phi);
+  const z = ((Math.pow(minorAxis, 2) * radius) /
+    Math.pow(majorAxis, 2) + alt) * Math.sin(lambda);
+
+  return [x, y, z];
 }
 
 /**
- * Converts lla coordinates to local world coordinates.
- * @param {double} lat Latitude in degrees.
- * @param {double} lng Longitude in degrees.
- * @param {double} alt Altitude in meters.
- * @return {array} An array returning gps coordinates.
+ * Converts Earth-Centered, Earth-Fixed coordinates (ECEF) to
+ * East-North-Up coordinates (ENU).
+ * @param {Object} ecefPose The pose data point being converted to ENU.
+ * @param {Object} ecefOrigin Reference ECEF position, used as the origin
+ *     of the 3D world.
+ * @param {Object} origin Reference LLA position, used as the origin
+ *     of the 3D world.
+ * @return {array} An array containing x, y, z ENU coordinates.
  */
-function llaDegreeToLocal(lat, lng, alt) {
-  /**
-   * Subtracting each point by the first point starts the pose
-   * at coordinates (0, 0, 0). Each unit in the 3D scene is 4
-   * meters while the 6th decimal point in GPS coordinates represents
-   * .11 meters. This is why we multiple the lat/lng by 25000, an
-   * increment in the 6th decimal place equates to a 0.025 unit
-   * change in our 3D space. Since altitude is already represented in
-   * meters, we simply divide by 4 to adjust for our 1:4 unit ratio.
-   */
-  const x = (lng - pose[0].lng) * 25000 * poseTransform.scale;
-  const y = (alt - pose[0].alt) / 4;
-  const z = (lat - pose[0].lat) * 25000 * -poseTransform.scale;
-  return [x, y, z];
+function ecefToEnu(ecefPose, ecefOrigin, origin) {
+  // ECEF to ENU equation: https://www.mathworks.com/help/map/ref/ecef2enu.html
```

Comment:
I didn't find the equation, could you point out where the equation is?

Reply:
Sorry, I accidentally pasted another website I was referencing. http://www.hydrometronics.com/downloads/Ellipsoidal%20Orthographic%20Projection.pdf 
This is the real link, the matrix algebra is on page 15.